packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m

```diff
@@ -168,6 +169,8 @@ - (void)launchImagePicker:(EXImagePickerTarget)target
     }
 
     self.picker.mediaTypes = [self convertMediaTypes:self.options[@"mediaTypes"]];
+
+    self.picker.videoMaximumDuration = [[self.options objectForKey:@"videoMaxDuration"] doubleValue];
```

Comment:
I'm a little concerned here 🤔 
Are you sure that passing `0` (the `unlimited` here) is forcing `UIImagePickerController` to record with no limit?
https://developer.apple.com/documentation/uikit/uiimagepickercontroller/1619146-videomaximumduration
Moreover I see that `allowsEditing` has to be `YES` in order to this `videoMaximumDuration` to take any effect (https://stackoverflow.com/questions/17375249/uiimagepickercontroller-not-respecting-video-maximum-duration-on-ipad/17375309) 🤔

Reply:
I did more research and testing, and I agree that setting it to 0 is a bit hacky and can give weird behavior (and one another problem 😞  ) when `allowsEditing` is true on iOS. The SO question about iPad you linked is only about if to show trimmer or not. I tested it deeper:

 - #### `alowsEditing = false`
    When limit set to 0, I can record video longer than 10 minutes and use it. No trimmer is shown.
   
  <img src="https://user-images.githubusercontent.com/278340/89008941-dfbc9300-d30b-11ea-8162-028d7edf49d6.jpeg" height="350"/>

   When limit is set to some value, video automatically stops recording after that time. No trimmer is shown.
  
 <img src="https://user-images.githubusercontent.com/278340/89009143-3c1fb280-d30c-11ea-9d05-526b0f0bed46.jpeg" height="350"/>

---

- ####  `allowsEditing = true`

 Here the problems begin. When I set any limit, recording also stops and I can trim my video. However, an ugly message is shown at the bottom:

<img src="https://user-images.githubusercontent.com/278340/89009646-3d051400-d30d-11ea-9a61-42915f8dc833.jpeg" height="350" />

This ugly message, _of course,_ isn't easily configurable, but someone found a way [on SO](https://stackoverflow.com/questions/40317259/video-too-long-title-alert-shown-by-uiimagepickercontroller).

When limit is set to 0 (no-limit) - this message is shown each time I end recording, regardless of video duration. It is probably related to another [`UIVideoEditorController.videoMaximumDuration`](https://developer.apple.com/documentation/uikit/uivideoeditorcontroller/1622338-videomaximumduration), which is probably set to the same as in `UIImagePickerController`.

Video editor docs say:
> The system-enforced maximum duration for a video recording is 10 minutes; you can set this value to 10 minutes or less. The default value for this property is also 10 minutes.

It seems that we cannot support videos longer than 10 minutes with `allowsEditing=true` 🙄 

Any ideas, suggestions what to do with it?

Reply:
Thank you for this deep investigation 💪 That helps a lot 😊 
Let's separate what should be done into two cases (depending on the `allowsEditing` flag being set or not):

1. `allowsEditing = false` (default value)
- I believe the default behaviour should be `no-limit` on both platforms (it's easily achievable form what I understand),
- If on both platforms passing explicit `0` value gives us this `no-limit` behaviour then I have no position to disagree and propose other approach 🤷 Maybe to judge such scenario in our docs, let's add additional note: `Both iOS and Android platforms consume 0 value as no-limit option.`

2. `allowsEditing = true` 
- How hard it is to customize this error message? (If it's fairly easy maybe we can do sth about it?)
- Let's add additional note into docs for iOS regarding this 10-minutes platform enforced limit and maybe reject on iOS if the user provides both `allowsEditing=true` and `maxDuration > 600`?

I summon @byCedric to provide some other point of view 😉

Reply:
I fully agree. Using sensible defaults is always a good one to do, even when it's different on platforms. If this is a limitation from Apple then let's make the most out of it and document that limitation. When this limitation is really hard to overcome, we might be able to split the config for android and ios. So developers can use different settings on both platforms?

Reply:
Ok, so IMO we can do it this way:
- `allowsEditing = false` - default to 0 on both platforms. If any problems would ever happen on iOS _(setting to 0 is a kind of hack after all)_, we can then fallback to 10 min limit on that platform.

- `allowsEditing = true` - defaults to 0 on Android and.. What about iOS?
  - We can set iOS default to 600 (max) and give a note in docs.
  - When user explicitly sets `maxDuration > 600` - we can reject, that's no problem.
  - When user sets it explicitly to 0 - we set it to 600: `if(maxDuration == 0) { maxDuration = 600; }` But should the user be warned somehow? If so, how?

_Also, as I am recently checking for corner cases for such options - negative values should be treated as no-limit or throw?_ 🤷 😉 

> How hard it is to customize this error message? (If it's fairly easy maybe we can do sth about it?)

I haven't tested it yet, but if [this answer](https://stackoverflow.com/a/54302086) works, it is just a category for `NSBundle`.

Reply:
- Upon negative values, let's reject (check that on JS side)
- `allowsEditing = true, maxDuration = 0 (no-limit), platform = iOS` (under the hood `maxDuration = 600`) - document it and that should do the thing 👍 
- if I understand correctly having above in place would not trigger this nasty error message from UIVideoEditorController? 🤔 If so, problem solved ✅ If no, we can try this approach from SO 💪

Reply:
> if I understand correctly having above in place would not trigger this nasty error message from UIVideoEditorController? 🤔 If so, problem solved ✅ If no, we can try this approach from SO 💪

I'm afraid we'll have to deal with it anyway. This message can be triggered (and this is expected behavior) when user sets the limit explicitly (e.g. to 30s) and when maximum duration is exceeded.

Reply:
Ok, let's do it then, shall we? 😊

Reply:
Unless it's rejected by Apple 🙄  cc @lukmccall